op_crates/fetch/lib.rs

```diff
@@ -33,34 +33,58 @@ use std::rc::Rc;
 
 pub use reqwest; // Re-export reqwest
 
+/// Execute this crates' JS source files.
 pub fn init(isolate: &mut JsRuntime) {
   let manifest_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
   let files = vec![
-    manifest_dir.join("01_fetch_util.js"),
-    manifest_dir.join("03_dom_iterable.js"),
-    manifest_dir.join("11_streams.js"),
-    manifest_dir.join("20_headers.js"),
-    manifest_dir.join("26_fetch.js"),
+    (
+      manifest_dir.join("01_fetch_util.js"),
+      include_str!("01_fetch_util.js"),
+    ),
+    (
+      manifest_dir.join("03_dom_iterable.js"),
+      include_str!("03_dom_iterable.js"),
+    ),
+    (
+      manifest_dir.join("11_streams.js"),
+      include_str!("11_streams.js"),
+    ),
+    (
+      manifest_dir.join("20_headers.js"),
+      include_str!("20_headers.js"),
+    ),
+    (
+      manifest_dir.join("26_fetch.js"),
+      include_str!("26_fetch.js"),
+    ),
   ];
   // TODO(nayeemrmn): https://github.com/rust-lang/cargo/issues/3946 to get the
   // workspace root.
   let display_root = manifest_dir.parent().unwrap().parent().unwrap();
-  for file in files {
-    println!("cargo:rerun-if-changed={}", file.display());
+  for (file, source_code) in files {
     let display_path = file.strip_prefix(display_root).unwrap();
     let display_path_str = display_path.display().to_string();
-    isolate
-      .execute(
-        &("deno:".to_string() + &display_path_str.replace('\\', "/")),
-        &std::fs::read_to_string(&file).unwrap(),
-      )
-      .unwrap();
+    let url = "deno:".to_string() + &display_path_str.replace('\\', "/");
+    isolate.execute(&url, source_code).unwrap();
```

Comment:
Do you need to generate a string at runtime to get the full path of the file? Would not it be enough to have a static &str with crate name plus file name, like `deno_fetch/01_fetch_util.js`?

If so, file name at lines 40-59 could be generated at compile time.

Reply:
Yes - it's a little annoying to have the paths hardcoded, but I agree that would be simpler. Updated.